### ISSUE TITLE
Update OpenAI API key

### ISSUE THREAD

**Issue by AlexanderEkdahl** [zed-industries/zed#4408]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Create new OpenAI key
2. Provide the above key to Zed
3. Revoke the OpenAI key
4. Zed Assistant panel no longer works with no way of providing a new OpenAI API key

The original feature announcement post mentions that the token is stored in my Keychain but I am not able to find it.

### Environment

Zed: v0.108.2 (stable)
OS: macOS 14.0.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="491" alt="Screenshot 2023-10-22 at 10 33 39 AM" src="https://github.com/zed-industries/community/assets/1409734/34f91d6c-05cf-49cd-af7e-adab25039a18">

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by antimora** [comment]
"assistant: reset key" is not there any more. Is it broken now?

**Comment by mrnugget** [comment]
It's still there. But the action only shows up when the assistant panel is opened _and focused_. I just updated the docs on that (and also added a note that you need credits for the key)

https://zed.dev/docs/assistant-panel

**Comment by caldempsey** [comment]
I genuinely can't find this on macbook @mrnugget . Also having this issue on Mac too https://github.com/zed-industries/zed/issues/6719. Can you check to see if the DEB package got updated?

**Comment by confiq** [comment]
> It's still there. But the action only shows up when the assistant panel is opened and focused. I just updated the docs on that (and also added a note that you need credits for the key)

> https://zed.dev/docs/assistant-panel

As this link do not work and I could not find this in the menu, I've used mac "keychain access" and deleted old key. 

Hope it helps someone

**Comment by mrnugget** [comment]
It's now here: https://zed.dev/docs/language-model-integration

And we now have a dedicated configuration view:
![screenshot-2024-08-12-08 45 16@2x](https://github.com/user-attachments/assets/15feb3db-2b91-4140-a115-080aa4912d47)

**Comment by confiq** [comment]
I did search for it, it took me around 1 hour! In version 0.142.2 this does not exist in the menu:

<img width="636" alt="image" src="https://github.com/user-attachments/assets/4a47b866-5a08-4951-adbe-c9ece1d6f3d6">


Thanks for looking at this 🙏

**Comment by mrnugget** [comment]
Yeah, it requires 0.148.0, which is our current Preview release (which is what I wrote above). It gets into stable on Wednesday.

**Comment by mrnugget** [comment]
I'm not sure I follow. Doesn't this work for you?


https://github.com/user-attachments/assets/a05d867e-817b-494b-9b21-ff356e4bdbf1

**Comment by mrnugget** [comment]
@D9J9V I misunderstood, sorry! The environment variable `OPENAI_API_KEY` can *not* be reset from within Zed, but you need to reset it outside, before starting Zed.

**Comment by confiq** [comment]
can confirm that on `0.148.1` this works perfect for me!
